extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/QuarkusSimplifiedIdentifierGeneratorFactory.java

```diff
@@ -16,95 +25,175 @@
+                    "supply a org.hibernate.id.factory.spi.GenerationTypeStrategyRegistration Java service");
+            final IdentifierGeneratorStrategyProvider idGeneratorStrategyProvider = serviceRegistry
+                    .getService(StrategySelector.class)
+                    .resolveStrategy(IdentifierGeneratorStrategyProvider.class, providerSetting);
+            for (Map.Entry<String, Class<?>> entry : idGeneratorStrategyProvider.getStrategies().entrySet()) {
+                @SuppressWarnings({ "rawtypes", "unchecked" })
+                Class<? extends IdentifierGenerator> generatorClass = (Class) entry.getValue();
+                register(entry.getKey(), generatorClass);
+            }
+        }
+    }
+
+    //Same-as-upstream - but made public
     public void register(String strategy, Class generatorClass) {
-        LOG.debugf("Registering IdentifierGenerator strategy [%s] -> [%s]", strategy, generatorClass.getName());
-        final Class previous = generatorStrategyToClassNameMap.put(strategy, generatorClass);
-        if (previous != null) {
-            LOG.debugf("    - overriding [%s]", previous.getName());
+        ID_GEN_FAC_LOGGER.debugf("Registering IdentifierGenerator strategy [%s] -> [%s]", strategy, generatorClass.getName());
+        final Class<?> previous = legacyGeneratorClassNameMap.put(strategy, generatorClass);
+        if (previous != null && ID_GEN_FAC_LOGGER.isDebugEnabled()) {
+            ID_GEN_FAC_LOGGER.debugf("    - overriding [%s]", previous.getName());
         }
     }
 
-    @Override
-    public Dialect getDialect() {
-        return dialect;
+    @Override //Same-as-upstream
+    public IdentifierGenerator createIdentifierGenerator(
+            GenerationType generationType,
+            String generatedValueGeneratorName,
+            String generatorName,
+            JavaType<?> javaType,
+            Properties config,
+            GeneratorDefinitionResolver definitionResolver) {
+        final GenerationTypeStrategy strategy = generatorTypeStrategyMap.get(generationType);
+        if (strategy != null) {
+            return strategy.createIdentifierGenerator(
+                    generationType,
+                    generatorName,
+                    javaType,
+                    config,
+                    definitionResolver,
+                    serviceRegistry);
+        }
+        throw new UnsupportedOperationException("No GenerationTypeStrategy specified");
     }
 
-    @Override
-    public void setDialect(Dialect dialect) {
-        //This is all commented out in the original code in Hibernate ORM as well
+    @Override //Same-as-upstream
+    public Dialect getDialect() {
+        if (dialect == null) {
+            dialect = serviceRegistry.getService(JdbcEnvironment.class).getDialect();
+        }
+        return dialect;
     }
 
+    //Different than upstream: ignore all complexity related to it having generators
+    //managed by a bean container.
```

Comment:
Have you considered `new StandardIdentifierGeneratorFactory(serviceRegistry, /* ignoreBeanContainer = */ true)`? You might be able to get rid of all the copied code...

Alternatively you can set `hibernate.cdi.extensions` to `false`, but that also affects `DefaultRevisionInfoGenerator`, not sure if that's what you want.

Reply:
That alternative has some problems, I have a (sadly local) list of very minor issues to report upstream (they're not urgent as they won't affect anyone else other than Quarkus users, for which I've bypassed the problem with this and similar overrides)